crypto/merkletrie/cache.go

```diff
@@ -438,10 +441,18 @@ func (mtc *merkleTrieCache) encodePage(nodeIDs map[storedNodeIdentifier]*node) [
 	return serializedBuffer[:walk]
 }
 
-// evict releases the least used pages from cache until the number of elements in cache are less than cachedNodeCountTarget
+// evict releases the least used pages from cache until the number of elements in cache are less than cachedNodeCountTarget.
+// the root element page is being moved to the front so that it would get flushed last.
 func (mtc *merkleTrieCache) evict() (removedNodes int) {
 	removedNodes = mtc.cachedNodeCount
-
+	// check the root element ( if there is such ), and give it a higher priority, since we want
+	// to release the page with the root element last.
+	if mtc.mt.root != storedNodeIdentifierNull {
+		rootPage := uint64(mtc.mt.root) / uint64(mtc.nodesPerPage)
+		if element, has := mtc.pagesPrioritizationMap[rootPage]; has && element != nil {
```

Comment:
I am not convinced why this change is needed.
If the reason for this is because we frequently use the root node page, it will already be taken care of by the least-recently-used eviction policy implemented by prioritizeNode, and keep this page in the cache. 

If this page is evicted frequently, could it be that there is a bug somewhere else causing it to happen?

Reply:
The use case here is a call to `trie.Evict(true)`, which would internally call `cache.commit()` followed by `cache.evict()`. 
When we're entering commit(), the root node page would be on the top. 
While executing `commit()`, we're trying to calculate the hashes of the various updated nodes. In many cases, we won't have the needed child node in memory, and we'll need to load it from disk. When we do this, that page is pushed to the front of the list. The update of the node hash value doesn't update its page position on the list.

Later on, when we get to call `evict()`, we find that the root page need to be evicted.
I have some bigger planned changes for this, which would hopefully eliminate the source of the problem - but I'd like to have a simple solution first, and address the "better" fix later on.